pkg/policyviolation/builder.go

```diff
@@ -0,0 +1,80 @@
+package policyviolation
+
+import (
+	"fmt"
+
+	kyverno "github.com/nirmata/kyverno/pkg/api/kyverno/v1"
+	client "github.com/nirmata/kyverno/pkg/dclient"
+)
+
+// Builder builds Policy Violation struct
+// this is base type of namespaced and cluster policy violation
+type Builder interface {
+	generate(info Info) []kyverno.PolicyViolation
+	build(policy, kind, namespace, name string, rules []kyverno.ViolatedRule) *kyverno.PolicyViolation
+}
+
+type pvBuilder struct {
+	// dynamic client
+	dclient *client.Client
+}
+
+func newPvBuilder(dclient *client.Client) *pvBuilder {
+	pvb := pvBuilder{
+		dclient: dclient,
+	}
+	return &pvb
+}
+func (pvb *pvBuilder) generate(info Info) []kyverno.PolicyViolation {
+	// https://github.com/nirmata/kyverno/issues/535
+	var owners []kyverno.ResourceSpec
+	// get the owners if the resource is blocked or
+	// the resource does not have a name assigned yet(uses generateName)
+	if info.Blocked || info.Resource.GetName() == "" {
```

Comment:
When `info.Resource.GetName() == ""`, assume it's a pod created by podcontroller, and for the first creation request, it doesn't come with the name, so here we are generating a violation on resource owner even if it's not blocked?

Reply:
this is related to https://github.com/nirmata/kyverno/issues/535, as nothing has been decided. so adding as a placeholder.

Reply:
Can we remove `info.Resource.GetName() == ""` condition here? As this would create incorrect violations on resourceOwner.

Reply:
ok will remove, but by when will be have the solution for #535 ?
As the current implementation works but is incorrect.

Reply:
removed the fix and jst left the placeholder.

Reply:
> ok will remove, but by when will be have the solution for #535 ?
> As the current implementation works but is incorrect.

Let's merge this PR first and move the discussion to 535.